java-checks/src/main/java/org/sonar/java/checks/xml/spring/SingleConnectionFactoryCheck.java

```diff
@@ -19,36 +19,43 @@
  */
 package org.sonar.java.checks.xml.spring;
 
+import java.util.stream.IntStream;
+
 import javax.xml.xpath.XPathExpression;
 import org.sonar.check.Rule;
 import org.sonarsource.analyzer.commons.xml.XmlFile;
 import org.sonarsource.analyzer.commons.xml.checks.SimpleXPathBasedCheck;
+import org.w3c.dom.NamedNodeMap;
 import org.w3c.dom.Node;
-import org.w3c.dom.NodeList;
+
 
 @Rule(key = "S3438")
 public class SingleConnectionFactoryCheck extends SimpleXPathBasedCheck {
 
   private XPathExpression singleConnectionFactoryBeansExpression = getXPathExpression("beans/bean[@class='org.springframework.jms.connection.SingleConnectionFactory']");
-  private XPathExpression reconnectOnExceptionPropertyValueExpression = getXPathExpression("property[@name='reconnectOnException' and value='true']");
+  private XPathExpression reconnectOnExceptionPropertyExpression = getXPathExpression("property[@name='reconnectOnException']");
+  private XPathExpression valueExpression = getXPathExpression("value[text()='true']");
 
   @Override
   public void scanFile(XmlFile file) {
     evaluateAsList(singleConnectionFactoryBeansExpression, file.getNamespaceUnawareDocument()).forEach(bean -> {
-      if (!hasPropertyAsAttribute(bean) && !hasPropertyAsChild(bean)) {
+      if (!hasAttributeValue(bean, "p:reconnectOnException") && !hasPropertyAsChild(bean, reconnectOnExceptionPropertyExpression)) {
```

Comment:
Is the `p:` really needed here?

Reply:
yes it is required because its on the bean element not on the property element

Reply:
I see, thanks for the clarification.